panic: invalid argument to Intn in ipam test

https://travis-ci.org/weaveworks/weave/builds/76423596
```
ok  	github.com/weaveworks/weave/ipam/paxos	2.985s
Ring [10.0.0.0, 10.0.0.255)
  10.0.0.0 -> 01:00:00:00:02:00 (v0)
--- FAIL: TestFuzzRingHard-4 (0.00s)
panic: invalid argument to Intn [recovered]
	panic: invalid argument to Intn
```

--- a/ipam/ring/ring_test.go
+++ b/ipam/ring/ring_test.go
@@ -709,6 +709,9 @@ func TestFuzzRingHard(t *testing.T) {
 				ringsWithEntries = append(ringsWithEntries, ring)
 			}
 		}
+		if len(ringsWithEntries) == 0 { // no rings have entries.
+			return
+		}
 		ring1 := ringsWithEntries[rand.Intn(len(ringsWithEntries))]
 		ring2index, _, ring2 := randomPeer(-1)
 		common.Log.Debugf("%s: 'Gossiping' to %s", ring1.Peer, ring2.Peer)

